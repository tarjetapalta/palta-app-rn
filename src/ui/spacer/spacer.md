Basic spacer component with size 'm' height '24px':

```jsx
<Spacer />
```

Spacer small 's' height '16px':

```jsx
<Spacer size="s" />
```

Spacer extra small 'xs' height '8px':

```jsx
<Spacer size="xs" />
```

Spacer large 'l' height '48px':

```jsx
<Spacer size="l" />
```

Spacer extra large 'xl' height '64px':

```jsx
<Spacer size="xl" />
```
