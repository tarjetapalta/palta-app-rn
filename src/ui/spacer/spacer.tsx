import React, { FunctionComponent } from 'react';

import { KeySizes } from '@theme/spacers';
import styled from '@theme/styled-components';

const SpacerWrapper = styled.View<{ size?: KeySizes }>`
  height: ${({ size, theme }) =>
    size ? theme.spacers[size] : theme.spacers.m};
`;

export const Spacer: FunctionComponent<{ size?: KeySizes }> = ({
  /** Sizes height to separate components */
  size,
}) => <SpacerWrapper size={size} />;
