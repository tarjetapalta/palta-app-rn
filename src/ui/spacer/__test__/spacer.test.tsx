import React from 'react';
import renderer from 'react-test-renderer';

import WithTheme from '../../../theme/context';
import { Spacer } from '../spacer';

describe('spacer component', () => {
  it('basic spacer component with size "m" height "24px"', () => {
    const result = renderer
      .create(
        <WithTheme>
          <Spacer />
        </WithTheme>,
      )
      .toJSON();

    expect(result).toMatchSnapshot();
  });
});
