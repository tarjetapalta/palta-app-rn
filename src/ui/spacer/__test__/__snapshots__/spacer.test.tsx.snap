// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`spacer component basic spacer component with size "m" height "24px" 1`] = `
<View
  style={
    Array [
      Object {
        "height": 24,
      },
    ]
  }
/>
`;
